### DISCUSSION TITLE
pdnsutil - why no delete-record or remove-record?

### DISCUSSION THREAD

**Discussion by bradley-baker** [PowerDNS/pdns#13297] [Q&A] [answered]
- Program: Authoritative
 - Issue type: Feature request

### Short description
The PowerDNS nameserver pdnsutil utility has an add-record, but no remove-record. How can we remove records programmatically for many domains at once? There should pdnsutil remove-record or pdnsutil delete-record command

### Usecase
I need to delete a large number of old name server (NS) records from over a hundred zones. Deleting them one by one would be very tedious.

### Description
I should be able to run a command such as:
pdnsutil remove-record mydomain.com @ NS dns2.mydomain.com

**Comment by Habbie** [comment]
Is `pdnsutil replace-rrset` a useful alternative?

**Comment by bradley-baker** [comment]
This does seem to work. Thank you! 

```
pdnsutil replace-rrset somedomain.org . NS ns1.ourdnsserver.org ns2.ourdnsserver.org
```

It would still be more intuitive if a create-record, update-record, and delete-record subcommand existed in my opinion, but thank you again so much. You just saved me hours of effort :)

**Comment by bradley-baker** [comment]
For anyone else trying to do this, this is how you do it for SOA records. Note the single quotes which are important:

```
pdnsutil replace-rrset domain.org . SOA 'ns1.ourdomain.org sysadmin.ourdomain.org 2023092502 10800 3600 604800 3600'
```

**Comment by bradley-baker** [comment]
@Habbie I may have another problem with the lack of a delete subcommand. 

We're migrating from a DNS server that doesn't support ALIAS/CNAME records for the apex/root domain. Now that we're on power DNS we want to switch from A records over to ALIAS/CNAME records so that we only have an IP address in one location. And we need to do this for hundreds of zones.

I can bulk-add ALIAS records like this:
```
pdnsutil add-record ourdomain.org @ 3600 ALIAS   www.ourdomain.org
```

But how do I remove all the old A records like this:
```
ourdomain.org    3600    IN      A       1.2.3.4
```

Can `pdnsutil replace-rrset` be used in this scenario since we're switching from A to ALIAS records?

**Reply by Habbie** [reply, marked as answer]
you can use `delete-rrset`

**Reply by bradley-baker** [reply]
This works:

```
pdnsutil delete-rrset abspritz.org @ A
```

Sorry for being a pain in the butt and thank you. I overlooked delete-rrset.

**Comment by Habbie** [comment]
I'm moving this to discussions, then I'll respond.

**Comment by bradley-baker** [comment]
Thank you
